tasks/gointegrationtest.py

```diff
@@ -0,0 +1,212 @@
+)
+
+CLUSTER_AGENT = IntegrationTestsConfig(
+    name="Cluster Agent",
+    go_build_tags=get_default_build_tags(build="cluster-agent") + ["docker", "test"],
+    tests=[
+        IntegrationTest(prefix="./test/integration/util/kube_apiserver"),
+        IntegrationTest(prefix="./test/integration/util/leaderelection"),
+    ],
+)
+
+TRACE_AGENT = IntegrationTestsConfig(
+    name="Trace Agent",
+    go_build_tags=get_default_build_tags(build="test"),
+    tests=[IntegrationTest(prefix="./cmd/trace-agent/test/testsuite/...")],
+    env={"INTEGRATION": "yes"},
+)
+
+
+def containerized_integration_tests(
+    ctx: Context,
+    integration_tests_config: IntegrationTestsConfig,
+    race=False,
+    remote_docker=False,
+    go_mod="readonly",
+    timeout="",
+):
+    test_args = {
+        "go_mod": go_mod,
+        "go_build_tags": " ".join(integration_tests_config.go_build_tags),
+        "race_opt": "-race" if race else "",
+        "exec_opts": "",
+        "timeout_opt": f"-timeout {timeout}" if timeout else "",
+    }
+
+    # since Go 1.13, the -exec flag of go test could add some parameters such as -test.timeout
+    # to the call, we don't want them because while calling invoke below, invoke
+    # thinks that the parameters are for it to interpret.
+    # we're calling an intermediate script which only pass the binary name to the invoke task.
+    if remote_docker:
+        test_args["exec_opts"] = f"-exec \"{os.getcwd()}/test/integration/dockerize_tests.sh\""
+
+    go_cmd = 'go test {timeout_opt} -mod={go_mod} {race_opt} -tags "{go_build_tags}" {exec_opts}'.format(**test_args)  # noqa: FS002
+
+    for it in integration_tests_config.tests:
+        if it.dir:
+            with ctx.cd(f"{it.dir}"):
+                ctx.run(f"{go_cmd} {it.prefix}", env=integration_tests_config.env)
+        else:
+            ctx.run(f"{go_cmd} {it.prefix}", env=integration_tests_config.env)
+
+
+def dsd_integration_tests(ctx, race=False, remote_docker=False, go_mod="readonly", timeout=""):
+    """
+    Run integration tests for dogstatsd
+    """
+    if sys.platform == 'win32':
+        raise TestsNotSupportedError('DogStatsD integration tests are not supported on Windows')
+    containerized_integration_tests(
+        ctx,
+        DOGSTATSD,
+        race=race,
+        remote_docker=remote_docker,
+        go_mod=go_mod,
+        timeout=timeout,
+    )
+
+
+def dca_integration_tests(ctx, race=False, remote_docker=False, go_mod="readonly", timeout=""):
+    """
+    Run integration tests for cluster-agent
+    """
+    if sys.platform == 'win32':
+        raise TestsNotSupportedError('Cluster Agent integration tests are not supported on Windows')
+    containerized_integration_tests(
+        ctx,
+        CLUSTER_AGENT,
+        race=race,
+        remote_docker=remote_docker,
+        go_mod=go_mod,
+        timeout=timeout,
+    )
+
+
+def trace_integration_tests(ctx, race=False, go_mod="readonly", timeout="10m"):
```

Comment:
Are these methods really needed (as they nearly only differ by their 2nd argument?)

Reply:
Good point, removed them all in [27dd5cc](https://github.com/DataDog/datadog-agent/pull/31801/commits/27dd5ccfd1b48b90877a572131543fe2ad714391)!